src/pow.cpp

```diff
@@ -92,6 +92,8 @@ std::string Pow::__str__() const
     std::ostringstream o;
     if (is_a<Add>(*base_)) {
         o << "(" << *base_ << ")";
+    } else if (is_a<Mul>(*base_)) {
+        o << "(" << *base_ << ")";
```

Comment:
We definitely can. But I felt having separate conditions for different classes will make it easier if we need to add subconditions to any of the classes, in future.
On second thought, however, we can make the changes when the need arises.